### PR TITLE
[k8s] Disable autostop for controller on kubernetes

### PR DESCRIPTION
**Pull request by romilbhardwaj**
Skips autostop when using the jobs controller on Kubernetes.

This should be merged after #3377, since that PR will add support for `SERVICE_ACCOUNT`, which is helpful for running the controller on GKE exec based auth clusters. 

Tested (run the relevant ones):

- [x] Code formatting: `bash format.sh`
- [x] Tested manually with `sky jobs launch` with controller and job on Kubernetes
- [x] Managed job smoke tests `pytest tests/test_smoke.py --managed-jobs --kubernetes`

### CODE REVIEW

File: sky/backends/cloud_vm_ray_backend.py [hunk 1]
```diff
@@ -4045,6 +4053,11 @@ def set_autostop(self,
         # The core.autostop() function should have already checked that the
         # cloud and resources support requested autostop.
         if idle_minutes_to_autostop is not None:
+            # Skip auto-stop for Kubernetes clusters.
+            if isinstance(handle.launched_resources.cloud, clouds.Kubernetes):
+                logger.info('Auto-stop is not supported for Kubernetes '
```
Comment (Michaelvll) [inline sky/backends/cloud_vm_ray_backend.py:4058]: Should we `assert controller_utils.Controllers.from_name(handle.cluster_name) == controller_utils.Controllers.JOBS_CONTROLLER` to ensure we do not accidentally get into this code path.

**Review by Michaelvll — APPROVED** [review]
Thanks @romilbhardwaj! LGTM.

File: sky/backends/cloud_vm_ray_backend.py [hunk 2]
```diff
@@ -1991,9 +1991,17 @@ def provision_with_retries(
                     cloud_user = None
                 else:
                     cloud_user = to_provision.cloud.get_current_user_identity()
+
+                requested_features = self._requested_features.copy()
+                # Skip stop feature for Kubernetes jobs controller.
+                if isinstance(to_provision.cloud, clouds.Kubernetes) and controller_utils.Controllers.from_name(
+                        cluster_name) == controller_utils.Controllers.JOBS_CONTROLLER:
+                    requested_features.remove(
```
Comment (Michaelvll) [inline sky/backends/cloud_vm_ray_backend.py:1999]:
```suggestion
                    assert clouds.CloudImplementationFeatures.STOP in requested_features, requested_features
                    requested_features.remove(
```

**Comment by romilbhardwaj** [comment]
Thanks @Michaelvll! Tested with `pytest tests/test_smoke.py --managed-jobs --kubernetes` on GKE cluster with this config.yaml:
```
kubernetes:
  remote_identity: SERVICE_ACCOUNT
jobs:
  controller:
    resources:
      cloud: kubernetes
      cpus: 8
      memory: 8
```

**Comment by Michaelvll** [comment]
> Thanks @Michaelvll! Tested with `pytest tests/test_smoke.py --managed-jobs --kubernetes` on GKE cluster with this config.yaml:
> 
> ```
> kubernetes:
>   remote_identity: SERVICE_ACCOUNT
> jobs:
>   controller:
>     resources:
>       cloud: kubernetes
>       cpus: 8
>       memory: 8
> ```

Awesome! Can we also test it without explicit specify the controller resources?

**Comment by romilbhardwaj** [comment]
Yes, smoke tests pass even without specifying controller resources! Kubernetes is automatically chosen for the controller if it has enough resources (8 CPU, 24 GB mem), verified with `kubectl get pods`.

File: sky/backends/cloud_vm_ray_backend.py [hunk 3]
```diff
@@ -1991,9 +1991,21 @@ def provision_with_retries(
                     cloud_user = None
                 else:
                     cloud_user = to_provision.cloud.get_current_user_identity()
+
+                requested_features = self._requested_features.copy()
+                # Skip stop feature for Kubernetes jobs controller.
+                if isinstance(to_provision.cloud, clouds.Kubernetes
+                             ) and controller_utils.Controllers.from_name(
+                                 cluster_name
+                             ) == controller_utils.Controllers.JOBS_CONTROLLER:
```
Comment (Michaelvll) [inline sky/backends/cloud_vm_ray_backend.py:2000]: nit: would be nice to add `()`
```suggestion
                if (isinstance(to_provision.cloud, clouds.Kubernetes
                             ) and controller_utils.Controllers.from_name(
                                 cluster_name
                             ) == controller_utils.Controllers.JOBS_CONTROLLER):
```

**Pull request merged**